src/vs/platform/windows/electron-main/window.ts

```diff
@@ -441,16 +443,45 @@ export class CodeWindow extends Disposable implements ICodeWindow {
 			return false;
 		};
 
-		const isRequestFromSafeContext = (details: Electron.OnBeforeRequestListenerDetails | Electron.OnHeadersReceivedListenerDetails): boolean => {
+		const isSvgRequestFromSafeContext = (details: Electron.OnBeforeRequestListenerDetails | Electron.OnHeadersReceivedListenerDetails): boolean => {
 			return details.resourceType === 'xhr' || isSafeFrame(details.frame);
 		};
 
+		const workbenchRootUrl = FileAccess.asBrowserUri('vs/', require).toString(true);
+		const isAllowedVsCodeFileRequest = (uri: URI, details: Electron.OnBeforeRequestListenerDetails) => {
+			// Allow vscode-file requests from main window
+			if (details.frame?.url.startsWith('vscode-file://vscode-app/')) {
```

Comment:
Maybe @deepak1556 has an idea if this check can be improved instead of probing on the URL.

Reply:
A frame is tracked in chromium by mapping the id of the process it belongs and a unique id within that process since there can be multiple frames in the same process. So always checking [frame.processId](https://github.com/electron/electron/blob/main/docs/api/web-frame-main.md#frameprocessid-readonly) and [frame.routingId](https://github.com/electron/electron/blob/main/docs/api/web-frame-main.md#frameroutingid-readonly) will be right way to identify a frame.

In this case, we want to compare the above properties of incoming frame with [this._win.webContents.mainFrame](https://github.com/electron/electron/blob/main/docs/api/web-contents.md#contentsmainframe-readonly) properties to allow requests for top level frame.

If we want to allow any frames from the workbench window process to use `vscode-file` protocol and only disallow for OOPIF used for extension webview, then checking only `frame.processId` is sufficient.